WcaOnRails/app/webpacker/components/Delegates/index.jsx

```diff
@@ -35,12 +36,28 @@ export default function Delegates() {
     [delegateGroups],
   );
 
-  const [activeRegion, setActiveRegion] = React.useState();
-  const [adminMode, setAdminMode] = React.useState(false);
+  const [hash, setHash] = useHash();
+
+  const SelectedComponent = React.useMemo(() => {
+    const selectedRegionIndex = delegateRegions.findIndex((region) => region.id === +hash);
+    if (selectedRegionIndex === -1) {
+      setHash(delegateRegions[0]?.id);
+      return () => null;
+    }
+    const selectedSection = delegateRegions[selectedRegionIndex];
+    if (selectedSection.component) {
+      return selectedSection.component;
+    }
+
+    return () => null;
+  }, [delegateRegions, hash, setHash]);
 
-  React.useEffect(() => {
-    setActiveRegion(delegateRegions?.[0]);
-  }, [delegateRegions]);
+  const activeRegion = React.useMemo(
+    () => delegateRegions.find((region) => region.id === +hash),
```

Comment:
Instead of find, we can use the selectedRegionIndex from above right?

Reply:
If we want to use `selectedRegionIndex`, it has to be outside `SelectedComponent`. Did you mean that?

Reply:
Yes, but according to the comment below, you won't be needing `SelectedComponent`. If that's the case, then this comment can be ignored.